conformance/ion_encoding/module/macro/trivial/invoke_tl.ion

```diff
@@ -0,0 +1,123 @@
+
+(ion_1_1 "Invalid bare reference"
+  (then "of undefined name"
+    (mactab (macro m () (noSuchMacro)))
+    (signals "No such macro: noSuchMacro"))
+ (then "cannot self-reference"
+   (mactab (macro m () (noSuchMacro)))
+   (signals "No such macro: noSuchMacro")))
+
+
+(ion_1_1 "Malformed macro references"
+  (each (mactab (macro _ () (':')))
+        (mactab (macro _ () ('::')))
+        (mactab (macro _ () (':1:')))
+        (mactab (macro _ () (':$ion:')))
+        (mactab (macro m () 0)
+                (macro _ () ('::m')))
+        (mactab (macro m () 0)
+                (macro _ () ('m:')))
+        (signals "Malformed macro-ref:")))
+
+
+(ion_1_1 "Invoking constant macros"
+  (each (mactab (macro m  ()  1)
+                (macro m1 ()  (m 1)))
+        (mactab (macro m  ()  1)
+                (macro m1 ()  (m 1 2)))
+        (mactab (macro m  ()  1)
+                (macro m1 ()  (m (;))))
+        (signals "Too many arguments"))
+  (then (mactab (macro m  ()  1)
+                (macro m1 ()  (m)))
+        (toplevel ('#$:m1'))
+        (produces 1)))
+
+
+(ion_1_1 "Local macros shadow system macros"
+  (mactab (macro values () "local")
+          (macro n () (values)))
+  (then (toplevel ('#$:$ion:values' "system")) // Precondition for this test
+        (produces "system"))
+  (then (toplevel ('#$:n'))
+        (produces "local")))
+
+
+(ion_1_1 "Qualified references"
+  (then (mactab (macro m (p*) (':$ion:values' 1 p 2))
+                (macro n (p*) (':$ion:1'      3 p 4)))
+        (toplevel ('#$:m' a b) ('#$:n' c d))
+        (produces 1 a b 2      3 c d 4))
+
+  (then "can reference a preceding module"
+    (encoding (module M (macro_table (macro m () "m")))
+              (module U (macro_table (macro u () [(':M:m'), (':M:0')])))  // No `use` needed!
+              (module V (use M)
+                        (macro_table (macro v () [(':M:m'), (':M:0')])))  // But it's allowed.
+              (macro_table U V))
+    (toplevel ('#$:u')   ('#$:v'))
+    (produces ["m", "m"] ["m", "m"])))
+
+
+(ion_1_1 "Local references"
+  (then (mactab (macro m () "m")
+                (macro n () [(':m'), (':0')]))
+        (toplevel ('#$:n'))
+        (produces ["m", "m"]))
+  (then "cannot self reference by name"
+    (mactab (macro m () (':m')))
+    (signals "No such macro declared in this module: :m"))
+  (then "cannot self reference by name"
+    (mactab (macro m () (':0')))
+    (signals "Invalid macro address: :0"))
+  (then "cannot forward reference by name"
+    (mactab (macro m () (':n'))
+            (macro n () "n"))
+    (signals "No such macro declared in this module: :n"))
+  (then "cannot forward reference by address"
+    (mactab (macro m () (':2'))
+            (macro n () "n")
+            (macro o () "o"))
+    (signals "Invalid macro address: :2"))
+  (then "cannot reference system macros"
+    (mactab (macro _ () (':values')))
+    (signals "No such macro declared in this module: :values"))
```

Comment:
Is this saying that system macros must have qualified names? Or that you have to specifically `load` the `$ion` module?

Reply:
Macro references of the form`:name` are _local references_ that can only address names bound inside this macro table.   (Thus the specific error message text).  Since the containing macro table's name is not in scope, this is how you disambiguate when the local name clashes with something from outside (including $ion)

Reply:
Okay. If this test is specifically for checking that local references are handled correctly, then the test description should be updated to make that clear.